src/compas_fab/robots/wrench.py

```diff
@@ -0,0 +1,395 @@
+    # access
+    # ==========================================================================
+
+    def __iter__(self):
+        return iter(list(self.force) + list(self.torque))
+
+    def __getitem__(self, item):
+        w = list(self)
+        return w[item]
+
+    # ==========================================================================
+    # representation
+    # ==========================================================================
+
+    def __repr__(self):
+        return "Wrench({0}, {1})".format(self.force, self.torque)
+
+    # ==========================================================================
+    # helpers
+    # ==========================================================================
+
+    def copy(self):
+        """Make a copy of this ``Wrench``.
+
+        Returns
+        -------
+        Wrench
+            The copy.
+        """
+        cls = type(self)
+        return cls(self.force.copy(), self.torque.copy())
+
+    # ==========================================================================
+    # operators
+    # ==========================================================================
+
+    def __mul__(self, n):
+        """Create a ``Wrench`` multiplied by the given factor.
+
+        Parameters
+        ----------
+        n : float
+            The multiplication factor.
+
+        Returns
+        -------
+        Wrench
+            The resulting new ``Wrench``.
+
+        """
+        return Wrench(self.force * n, self.torque * n)
+
+    def __add__(self, other):
+        """Return a ``Wrench`` that is the the sum of this ``Wrench`` and another wrench.
+
+        Parameters
+        ----------
+        other : wrench
+            The wrench to add.
+
+        Returns
+        -------
+        Wrench
+            The resulting new ``Wrench``.
+
+        """
+        return Wrench(
+            self.force + other.force,
+            self.torque + other.torque)
+
+    def __sub__(self, other):
+        """Return a ``Wrench`` that is the the difference between this ``Wrench`` and another wrench.
+
+        Parameters
+        ----------
+        other : wrench
+            The wrench to subtract.
+
+        Returns
+        -------
+        Wrench
+            The resulting new ``Wrench``.
+        """
+        return Wrench(
+            self.force - other.force,
+            self.torque - other.torque)
+
+    def __neg__(self):
+        """Return the negated ``Wrench``.
+
+        Returns
+        -------
+        Wrench
+            The negated ``Wrench``.
+        """
+        return Wrench(
+            self.force * -1.0,
+            self.torque * -1.0)
+
+    def __len__(self):
+        return 6
+
+    # ==========================================================================
+    # comparison
+    # ==========================================================================
+
+    def __eq__(self, other, tol=1e-05):
```

Comment:
For python 2 compatibility, you need to implement `def __ne__` as well (simply as `return not self.__eq__(other, tol)`)

Reply:
ok: implemented